Main/ripple.sh

```diff
@@ -0,0 +1,358 @@
+#!/bin/sh
+: '
+-------------------------------------------------------------------------------------
+|  Created by Angel Uniminin <uniminin@zoho.com> in 2019 under the terms of AGPLv3  |
+|                          Last Updated on 10th May, 2020                           |
+-------------------------------------------------------------------------------------
+'
+# Checking If Running [Script] as Root
+if ! [ "$(id -u)" = 0 ]; then
+   printf '%s\n' "Notice: Execute the Script as Root."
+   exit 1
+fi
+
+# FIXME: Proper/Better Way to do this.
+# Python 3.5 for peppy
+
+python3() {
+
+	python3_5() {
+		(
+			cd /usr/src || exit ; wget https://www.python.org/ftp/python/3.5.9/Python-3.5.9.tar.xz
+			tar -xvf Python-3.5.9.tar.xz ; cd Python-3.5.9 || exit
+			./configure --enable-loadable-sqlite-extensions --enable-optimizations ; make ; make install
+		)
+	}
+
+	# Python 3.6 for lets
+	python3_6() {
+		(
+			cd /usr/src || exit ; wget https://www.python.org/ftp/python/3.6.8/Python-3.6.8.tgz
+			tar -xvf Python-3.6.8.tgz ; cd Python-3.6.8 || exit
+			./configure --enable-optimizations ; make ; make install
+		)
+	}
+
+	python3_5
+	python3_6
+}
+
+# FIXME: check all dependencies, add/remove if needed for all supported systems.
+# Install necessary dependencies required for lets, pep.py, hanayo, go, old-frontend, mysql. (Used APT)
+dependencies() {
+	printf '%s\n' "Starting To Install Required/Necessary Dependencies..." ; sleep 2
+
+	if command -v apt >/dev/null; then
+		package_manager="apt"
+		printf '%s\n' "Using Package Manager: $package_manager"
+		"$package_manager" update ; "$package_manager" upgrade -y
+		"$package_manager" install build-essential gcc g++ \
+		git tmux nginx wget gnupg mysql-server redis-server nginx checkinstall cython \
+		php-fpm composer php7.0-mbstring php7.0-curl php-mysql -y
+
+		# Golang1.13+ for Hanayo & rippleapi
+		(
+			mkdir builds ; cd builds || exit
+			wget https://dl.google.com/go/go1.13.3.linux-amd64.tar.gz
+			tar -xvf go1.13.3.linux-amd64.tar.gz ; mv go /usr/local
+			export GOROOT=/root/go
+			export PATH=$GOPATH/bin:$GOROOT/bin:$PATH
+		)
+		# FIXME: it's ugly way to do. Provide me better way of this implementation
+		python3 ; "$package_manager" install python3-dev
+
+		printf '%s\n' "Done Installing all the necessary Dependencies with [$package_manager]" ; sleep 1
+
+	elif ! command -v apt >/dev/null; then
+    		printf '%s\n' "apt is not executable on this system!"
+
+	elif command -v pacman >/dev/null; then
+		package_manager="pacman"
```

Comment:
bad logic just use pacman unless you are expecting distro maintainer to change package manager which would break your script anyway